### ISSUE TITLE
Masking Support for Training Data  in Preprocessing

### ISSUE THREAD

**Issue by Laity2333** [WenjieDu/PyPOTS#595]
### Issue description

thank you for creating and sharing this valuable resource! 
While working with the PhysioNet-2012 dataset using your preprocessing pipeline, I noticed that explicit masking mechanisms for training data might not be currently implemented. This feature is crucial for training conditional imputation models.

**Comment by WenjieDu** [comment]
Hi, what do you mean by "explicit masking mechanisms for training data"?

**Comment by Laity2333** [comment]
In preprocessing, both the val_set and the test_set were masked proportionally, but the training set was not masked

**Comment by WenjieDu** [comment]
We include the masking procedure of training set into the forward process of models because different models may require different masking mechanisms. The validation set is preprocessed to bear artificial missingness because we need to ensure the training model is fairly evaluated for each epoch, hence the part for evaluation (i.e., the artificial missingness) should be fixed.

**Comment by Laity2333** [comment]
Thanks for your reply, it has effectively resolved my doubts.

**Comment by WenjieDu** [comment]
no problem
